### PR TITLE
Introduce the notion of default port for the specific drivers. In the…

### PR DESCRIPTION
**Pull request by davecramer**
… event that the port is not specified on the URL; we need a port to connect to. Changed server to host as server was being used in many places. Added a service file to allow the driver to be registered automatically

### CODE REVIEW

File: arcCommon/src/test/java/org/threadly/db/aurora/DelegateMockDriver.java [hunk 1]
```diff
@@ -35,8 +35,13 @@ public static Pair<MockDriver, DelegateAuroraDriver> setupMockDriverAsDelegate()
       public String getDriverName() {
         return "MockDriver";
       }
-      
-      @Override
+
+        @Override
```
Comment (jentfoo) [inline arcCommon/src/test/java/org/threadly/db/aurora/DelegateMockDriver.java:39]: Spacing looks off, please correct and make sure you're using spaces

**Review by jentfoo — CHANGES_REQUESTED** [review]
Changes overall look good, I left a few comments to fix and then I am happy to merge and release this change.  Thank you!

File: .idea/compiler.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        <processorPath useClasspath="false">
+          <entry name="$USER_HOME$/.gradle/caches/modules-2/files-2.1/org.openjdk.jmh/jmh-generator-annprocess/1.34/aa623312520ba3408bfab3e0983a3af2117ee381/jmh-generator-annprocess-1.34.jar" />
+          <entry name="$USER_HOME$/.gradle/caches/modules-2/files-2.1/org.openjdk.jmh/jmh-core/1.34/36d786a56ce4955779ae69d52be7081551490bb1/jmh-core-1.34.jar" />
+          <entry name="$USER_HOME$/.gradle/caches/modules-2/files-2.1/net.sf.jopt-simple/jopt-simple/5.0.4/4fdac2fbe92dfad86aa6e9301736f6b4342a3f5c/jopt-simple-5.0.4.jar" />
+          <entry name="$USER_HOME$/.gradle/caches/modules-2/files-2.1/org.apache.commons/commons-math3/3.2/ec2544ab27e110d2d431bdad7d538ed509b21e62/commons-math3-3.2.jar" />
+        </processorPath>
+        <module name="auroraArc.arcCommon.test" />
+      </profile>
+    </annotationProcessing>
+    <bytecodeTargetLevel target="16" />
```
Comment (jentfoo) [inline .idea/compiler.xml:16]: Currently we are targeting JDK 11.  I would either update the idea files to work for our styling and target level, or remove these additional files (similar errors show below).

File: .idea/jarRepositories.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="RemoteRepositoriesConfiguration">
+    <remote-repository>
+      <option name="id" value="central" />
+      <option name="name" value="Maven Central repository" />
+      <option name="url" value="https://repo1.maven.org/maven2" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="jboss.community" />
```
Comment (jentfoo) [inline .idea/jarRepositories.xml:10]: Should be unnecessary

File: arcCommon/src/test/java/org/threadly/db/aurora/DelegateMockDriver.java [hunk 4]
```diff
@@ -35,8 +35,13 @@ public static Pair<MockDriver, DelegateAuroraDriver> setupMockDriverAsDelegate()
       public String getDriverName() {
         return "MockDriver";
       }
-      
+
       @Override
+      public int getDefaultPort() {
```
Comment (jentfoo) [inline arcCommon/src/test/java/org/threadly/db/aurora/DelegateMockDriver.java:40]: Spacing still looks incorrect based off function below

**Review by jentfoo — CHANGES_REQUESTED** [review]
More changes requested.  If you feel I am being pedantic let me know and I will apply the updates myself before merging.  In general I like the change still.  Thank you!

File: .idea/compiler.xml (hunk 2 above)
Comment (davecramer) [reply .idea/compiler.xml:16]: Love intellij, and my lack of paying attention at 6am... didn't mean to push that.

File: .idea/jarRepositories.xml (hunk 3 above)
Comment (davecramer) [reply .idea/jarRepositories.xml:10]: same as above.

**Review by jentfoo — APPROVED** [review]
Looks good!  @davecramer I will go ahead and merge this now.

Are you using the driver from maven central, would a prompt public release be useful to you?  Otherwise I was going to take the time to update our driver and test dependencies before our next release.

**Comment by davecramer** [comment]
@jentfoo I'm testing it currently. I can use a snapshot release if you have one. Otherwise I can build it myself.
Thanks!

**Comment by jentfoo** [comment]
I attempted to release a snapshot but there was a few unit test failures I had missed.  My apologies, I have now corrected those issues: https://github.com/threadly/auroraArc/commit/4741fb052a88f41b02b30cf75c96c2ef182027cb

0.17-SNAPSHOT should also be available now, let me know if you run into any issues.  I will update the dependent drivers this weekend and release to maven central.  Thank you again for the help!

**Comment by davecramer** [comment]
> 0.17-SNAPSHOT should also be available now, let me know if you run into any issues. I will update the dependent drivers this weekend and release to maven central. Thank you again for the help!

my pleasure

**Pull request merged**